test/internal/kubernetes/utils.go

```diff
@@ -233,6 +233,24 @@ func WaitForPodsRunning(ctx context.Context, clientset *kubernetes.Clientset, na
 	return errors.Wrap(retrier.Do(ctx, checkPodIPsFn), "failed to check if pods were running")
 }
 
+func WaitForPodsDelete(ctx context.Context, clientset *kubernetes.Clientset, namespace, labelselector string) error {
+	podsClient := clientset.CoreV1().Pods(namespace)
+
+	checkPodsDeleted := func() error {
+		podList, err := podsClient.List(ctx, metav1.ListOptions{LabelSelector: labelselector})
+		if err != nil {
+			return errors.Wrapf(err, "could not list pods with label selector %s", labelselector)
+		}
+		if len(podList.Items) != 0 {
+			return errors.Errorf("%d pods still present", len(podList.Items))
```

Comment:
Would the verbosity of `podList.Items` give a wrong count from len?

Reply:
At least in my debugger, each pod is an item in the slice, so I think it's the right number